lms/services/moodle.py

```diff
@@ -45,6 +47,85 @@ def course_group_sets(self, course_id: int) -> list[dict]:
 
         return [{"id": g["id"], "name": g["name"]} for g in response]
 
+    def course_contents(self, course_id: int) -> list[dict]:
+        url = self._api_url(Function.GET_COURSE_CONTENTS)
+        response = self._http.post(url, params={"courseid": course_id}).json()
+        return response
+
+    def list_files(self, course_id: int):
+        contents = self.course_contents(course_id)
+        files = []
+
+        for topic in contents:
+            topic_name = topic["name"]
+
+            for module in topic["modules"]:
+                # Files can be at the top level modules
+                if module["modname"] == "resource" and module["modplural"] == "Files":
+                    files.extend(
+                        self._get_contents(module["contents"], parent=topic_name)
+                    )
+
+                # Or nested inside folders
+                elif module["modname"] == "folder":
+                    files.extend(
+                        self._get_contents(
+                            module["contents"], parent=topic_name + "/" + module["name"]
+                        )
+                    )
+                else:
+                    continue
+
+        return self._construct_file_tree(course_id, files)
+
+    @staticmethod
+    def _get_contents(contents, parent=None):
+        file_paths = []
+        for content in contents:
+            file_path = f"{parent}{content['filepath']}{content['filename']}"
+            file_paths.append(
+                {
+                    "path": file_path,
+                    "url": content["fileurl"],
+                    "updated_at": content["timemodified"] * 1000,
+                }
```

Comment:
We should probably start thinking on having a model in the backend that represents Files/Folders and wraps the logic to JSON-serialize it, instead of using raw dicts.

Something to consider for the future.